ssz-rs/benches/compute_proof.rs

```diff
@@ -0,0 +1,56 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion};
+use std::hint::black_box;
+
+use ssz_rs::{List, PathElement, Prove};
+use std::convert::TryFrom;
+
+/// Generate fake transactions to benchmark with.
+fn generate_dummy_transactions(size: usize) -> List<[u8; 32], 1000> {
+    let txs = (0..size).map(|i| [i as u8; 32]).collect::<Vec<_>>();
+    List::try_from(txs).expect("Failed to create transaction list")
+}
+
+/// Function to determine transaction indices based on the current size
+/// Returns first, half, and second last indexes.
+fn get_transaction_indices(size: usize) -> Vec<usize> {
+    if size == 0 {
+        vec![]
+    } else {
+        vec![0, size / 2, size - 1]
+    }
+}
+
+fn bench_prove(c: &mut Criterion) {
+    let sizes = [100, 300, 1000];
+
+    for &size in &sizes {
+        let dummy_transactions = generate_dummy_transactions(size);
+        let indices = get_transaction_indices(size);
+        let mut group = c.benchmark_group(format!("Prove Benchmark - size {}", size));
+        // Reduce sample size as benchmark for 1000 is unable to complete
+        // 100 samples in 5.0s on an M2 laptop.
+        group.sample_size(50);
```

Comment:
I'd reduce the sample size only when `size` is `1_000`, or I'd remove the large size at all, but up to you.